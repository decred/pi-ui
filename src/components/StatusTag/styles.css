.statusTagWrapper {
  padding: 0em 0.4em;
  height: 20px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 2px;
  border: 1px solid var(--color-gray);
}

.statusTagWrapper > span {
  font-family: var(--font-family-text);
  font-size: var(--font-size-small);
  line-height: var(--spacing-1);
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.statusTagWrapper > img {
  margin-right: 0.5rem;
}

.greenCheck {
  color: var(--color-primary-light);
  border-color: var(--color-primary-light);
}

.grayNegative {
  color: var(--color-gray);
  border-color: var(--color-gray);
}

.orangeNegativeCircled {
  color: var(--color-orange);
  border-color: var(--color-orange);
}

.bluePending {
  color: var(--color-primary);
  border-color: var(--color-primary);
}

.yellowTime {
  color: var(--color-yellow);
  border-color: var(--color-yellow);
}

.blackTime {
  color: var(--color-primary-dark);
  border-color: var(--color-primary-dark);
}

.blueNegative {
  color: var(--color-dark);
  border-color: var(--color-dark);
}
