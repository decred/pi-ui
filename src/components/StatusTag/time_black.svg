<svg xmlns="http://www.w3.org/2000/svg" width="8.544" height="8.639" viewBox="0 0 8.544 8.639"><g transform="translate(0 0)" opacity="0.9"><path d="M287.02,178.66a2.275,2.275,0,0,1-.376.269h0l-.215.215.806.752.215-.215c.107-.054.161-.161.269-.215l-.7-.806Zm-1.182,1.236h0l-.054.054h0a1.175,1.175,0,0,1-.161.269h0v.054h0c0,.054-.054.107-.054.161l1.021.376a2.878,2.878,0,0,1,.269-.591l-1.021-.322Zm-.537,1.612v.161h0v.591l1.075-.054a2.615,2.615,0,0,1,.054-.645l-1.129-.054Zm.054,1.72c.054.161.107.376.161.537h0v.054h0l.967-.484a2.621,2.621,0,0,1-.215-.591l-.914.484Zm1.612.86-.86.645.054.054h0l.161.161h0c.054,0,.054.054.107.054l.7-.806c-.107-.107-.215-.161-.269-.269.215.269.161.215.107.161Zm.967.86-.537.914a.41.41,0,0,0,.269.107h0a.764.764,0,0,0,.322.107l.322-1.021c0,.054-.161,0-.376-.107Zm1.236.376-.107,1.075a2.313,2.313,0,0,0,.86,0l-.107-1.075a2.59,2.59,0,0,1-.645,0Zm1.29-.107.322,1.021a2.534,2.534,0,0,0,.806-.322l-.484-.967c-.269.107-.484.215-.645.269Zm1.558-1.075c-.107.107-.215.269-.322.376l-.107.107.7.806c.054-.054.107-.107.161-.107h0a3.477,3.477,0,0,0,.43-.484l-.86-.7Zm.591-1.129c-.054.215-.161.43-.215.591l.967.484c.054-.107.107-.269.161-.376h0v-.054h0V183.6h0c0-.054.054-.161.054-.215l-.967-.376Zm.107-1.236a2.589,2.589,0,0,1,0,.645l1.075.107v-.484h0v-.054h0v-.054h0v-.215l-1.075.054Zm-.376-1.236a1.986,1.986,0,0,1,.269.591l1.021-.322c-.054-.161-.107-.269-.161-.43h0c-.054-.107-.107-.161-.161-.269l-.967.43Zm-.161-1.827-.645.86a3.071,3.071,0,0,1,.43.376l.054.054.806-.7v-.054h0a4.514,4.514,0,0,1-.645-.537Zm-1.558-.752-.269,1.021a1.926,1.926,0,0,1,.591.215l.484-.967a4.731,4.731,0,0,0-.806-.269Zm-1.72-.107.161,1.075a2.616,2.616,0,0,1,.645-.054l.054-1.075a4.917,4.917,0,0,0-.86.054Zm-1.612.591.537.914a1.987,1.987,0,0,1,.591-.269l-.376-1.021a2.621,2.621,0,0,0-.376.161h0a3.084,3.084,0,0,1-.376.215Z" transform="translate(-285.246 -177.8)" fill="#8997a5"/><g transform="translate(0 0)"><path d="M289.5,177.8a4.312,4.312,0,0,0-4.3,4.3,1.827,1.827,0,0,0,.054.537h1.075a1.827,1.827,0,0,1-.054-.537,3.008,3.008,0,0,1,.967-2.257,3.175,3.175,0,0,1,1.72-.914v2.633H287.35v1.075h2.687v-4.783c-.215,0-.376-.054-.537-.054Z" transform="translate(-285.2 -177.8)" fill="#091440"/></g></g></svg>