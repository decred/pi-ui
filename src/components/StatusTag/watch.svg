<svg xmlns="http://www.w3.org/2000/svg" width="8.672" height="8.673" viewBox="0 0 8.672 8.673">
  <g id="Symbol_591_2" data-name="Symbol 591 â€“ 2" transform="translate(-427.097 -342.219)">
    <rect id="XMLID_4_" width="0.734" height="3.093" transform="translate(431.092 343.739)" fill="#cfa03f"/>
    <rect id="XMLID_3_" width="0.734" height="2.376" transform="translate(431.829 346.261) rotate(90)" fill="#cfa03f"/>
    <path id="Path_184" data-name="Path 184" d="M102.436,51.14a3.6,3.6,0,1,1-3.6,3.6,3.6,3.6,0,0,1,3.6-3.6m0-.74a4.336,4.336,0,1,0,4.336,4.336,4.338,4.338,0,0,0-4.336-4.336Z" transform="translate(328.997 291.818)" fill="#cfa03f"/>
  </g>
</svg>
