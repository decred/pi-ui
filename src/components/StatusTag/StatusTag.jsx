import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.module.css";
import { classNames } from "../../utils";

import greenCheckIcon from "./green_check.svg";
import negativeIcon from "./negative.svg";
import negativeBlueIcon from "./negative_blue.svg";
import negativeCircleIcon from "./negative_circle.svg";
import pendingIcon from "./waiting.svg";
import timeYellowIcon from "./time_yellow.svg";
import timeBlackIcon from "./time_black.svg";
import timeBlueIcon from "./time_blue.svg";

const ICONS_MAP = {
  greenCheck: greenCheckIcon,
  grayNegative: negativeIcon,
  blueNegative: negativeBlueIcon,
  orangeNegativeCircled: negativeCircleIcon,
  bluePending: pendingIcon,
  yellowTime: timeYellowIcon,
  blackTime: timeBlackIcon,
  blueTime: timeBlueIcon,
};

const getIcon = (type) => ICONS_MAP[type];

// TODO: replace icons with SVG sprite file ones when we have them
const StatusTag = ({ type, text, className }) => {
  return (
    <span
      className={classNames(styles.statusTagWrapper, styles[type], className)}>
      <img src={getIcon(type)} alt={type} />
      <span>{text || type}</span>
    </span>
  );
};

StatusTag.propTypes = {
  type: PropTypes.oneOf(Object.keys(ICONS_MAP)),
  text: PropTypes.string,
  className: PropTypes.string,
};

StatusTag.defaultProps = {
  type: "grayNegative",
};

export default StatusTag;
