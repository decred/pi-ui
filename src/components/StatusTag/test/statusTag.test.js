import React from "react";
import StatusTag from "../StatusTag";
import { create } from "react-test-renderer";

describe("StatusTag component", () => {
  test("Matches the snapshot", () => {
    const statusTag = create(<StatusTag />);
    expect(statusTag.toJSON()).toMatchSnapshot();
  });
});
