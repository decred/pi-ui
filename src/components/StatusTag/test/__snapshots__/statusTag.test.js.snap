// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusTag component Matches the snapshot 1`] = `
<span
  className="statusTagWrapper grayNegative"
>
  <img
    alt="grayNegative"
    src=""
  />
  <span>
    grayNegative
  </span>
</span>
`;
